### PR TITLE
Indexed Entity X-Rays

### PR DESCRIPTION
**Pull request by markbastian**
Adds the ability to x-ray indexed entities. Primary logic is in `metabase.api.automagic-dashboards` and does the following:

- Adds an endpoint `GET "/model_index/:model-index-id/primary_key/:pk-id"` that takes the index and primary key of an indexed entity (i.e. a row in a table or model that has been indexed)
- Looks up the corresponding `ModelIndex`,`Card` (model), and `ModelIndexValue`
- Computes entities linked to this row via FK
- Generates a dashboard for each of these entities
- Merges the above dashboards into a single dashboard with each individual dashboard grouped onto tabs

### CODE REVIEW

File: src/metabase/automagic_dashboards/filters.clj [hunk 1]
```diff
@@ -78,7 +78,7 @@
 
 (defn- build-fk-map
   [fks field]
```
Comment (markbastian) [inline src/metabase/automagic_dashboards/filters.clj:80]: This is a bugfix that ensures that filters show up on entities with no fks.

**Comment by dpsutton** [comment]
I think there's a rough edge that we need to smooth out. For mbql queries, we will often find linked tables to x-ray, but not always. And for native queries, we will never find linked tables to x-ray unless they have identified the pk column with a column in the database.

image

When this scenario happens, we don't have anything to x-ray and we just return an error. We need to make sure this error is a polished state in the flow, and it includes enough information to understand why it happened, and what they can do to correctly link up their model.

**Comment by dpsutton** [comment]
For instance, I've created a native model on products (`select * from products`) named "native model on products without column info". Add an index marking the title as the label and id as the primary key.

image

When I go to the indexed x-ray I see the following:
image

Not ideal.

**Comment by markbastian** [comment]
I've added the following to address the no linked entities edge case.

boring

**Comment by markbastian** [comment]
Here's another weird case in which an index is added to native model (`SELECT * FROM PEOPLE`) and an index is added on address.

whatsup

**Comment by markbastian** [comment]
There were a couple issues causing the spinner issues.

- Complex 2-step joins that can actually be accomplished in a single filter
- Using tab logic when there is only one tab, which causes FE rendering problems

Both of these have been fixed. Here's an example of a chart that was having issues 
with both cases and is now fixed.

fixed

**Review by dpsutton — APPROVED** [review]

**Pull request merged**